mantidproject/mantid
# Reduce workspace history duplication

**Description of work.**
I accomplished two things. 1. To properly compare the uniqueness of piece of Algorithm history by comparing name and execution date to the nano second, 2. Ensure that it is possible to import and load from files the correct history even though it stores them to the second in current .nxs implementations, this is achieved by assuming the history is complete and correct and in order, it now increments the nanosecond counter by 1 until it is unique and can fit in the std::set

**To test:**
- Run a script that produces a large amount of history for different workspaces and then groups them together a good example is the GEM Focus scripts @SamJenkins1 
- Check the history of the output workspaces and see that their history is correct (may need to look at what your script is doing for this)
- Save out the Group that was made by the script into a nexus
- Load in that group again after clearing the ADS
- Check that the history of the workspaces loaded back in are the same as the previously saved ones

Fixes #23748  


Doesn't require release notes as this effects only the current dev-builds due it being a relatively recent change that causes history to record properly, when utilizing the ADSValidator on a string list of workspaces.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
